java/src/test/java/org/rocksdb/ReadOnlyTest.java

```diff
@@ -31,81 +31,51 @@ public void readOnlyOpen() throws RocksDBException {
-            ) {
-              assertThat(db2.get("key2".getBytes())).isNull();
-              assertThat(db2.get(readOnlyColumnFamilyHandleList.get(0),
-                  "key2".getBytes())).
-                  isNull();
-              cfDescriptors.clear();
-              cfDescriptors.add(
-                  new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY,
-                      newCfOpts2));
-              cfDescriptors.add(new ColumnFamilyDescriptor("new_cf2".getBytes(),
-                      newCf2Opts2));
-
-              final List<ColumnFamilyHandle> readOnlyColumnFamilyHandleList2
-                  = new ArrayList<>();
-              try (final RocksDB db3 = RocksDB.openReadOnly(
-                  dbFolder.getRoot().getAbsolutePath(), cfDescriptors,
-                  readOnlyColumnFamilyHandleList2)) {
-                try {
-                  assertThat(new String(db3.get(
-                      readOnlyColumnFamilyHandleList2.get(1),
-                      "key2".getBytes()))).isEqualTo("value2");
-                } finally {
-                  for (final ColumnFamilyHandle columnFamilyHandle :
-                      readOnlyColumnFamilyHandleList2) {
-                    columnFamilyHandle.close();
-                  }
-                }
-              }
-            } finally {
-              for (final ColumnFamilyHandle columnFamilyHandle :
-                  readOnlyColumnFamilyHandleList) {
-                columnFamilyHandle.close();
-              }
-            }
-          }
-        } finally {
-          for (final ColumnFamilyHandle columnFamilyHandle :
-              columnFamilyHandleList) {
-            columnFamilyHandle.close();
-          }
-        }
+    final List<ColumnFamilyHandle> readOnlyColumnFamilyHandleList = new ArrayList<>();
+    try (final RocksDB db = RocksDB.openReadOnly(
+             dbFolder.getRoot().getAbsolutePath(), cfDescriptors, readOnlyColumnFamilyHandleList)) {
+      assertThat(db.get("key2".getBytes())).isNull();
+      assertThat(db.get(readOnlyColumnFamilyHandleList.get(0), "key2".getBytes())).isNull();
+    } finally {
+      for (ColumnFamilyHandle handle : readOnlyColumnFamilyHandleList) {
+        handle.close();
+      }
+    }
+
+    cfDescriptors.clear();
+    cfDescriptors.add(
+        new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, new ColumnFamilyOptions()));
+    cfDescriptors.add(new ColumnFamilyDescriptor("new_cf2".getBytes(), new ColumnFamilyOptions()));
+    final List<ColumnFamilyHandle> readOnlyColumnFamilyHandleList2 = new ArrayList<>();
+    try (final RocksDB db = RocksDB.openReadOnly(dbFolder.getRoot().getAbsolutePath(),
+             cfDescriptors, readOnlyColumnFamilyHandleList2)) {
+      assertThat(new String(db.get(readOnlyColumnFamilyHandleList2.get(1), "key2".getBytes())))
+          .isEqualTo("value2");
+    } finally {
+      for (ColumnFamilyHandle handle : readOnlyColumnFamilyHandleList2) {
```

Comment:
In the latest codebase. Any column family handles created by a `RocksDB#open` call, will be auto-magically cleaned up in the corresponding `RocksDB#close` call.

So you may be able to benefit from that, and tidy this up further.